src/mono/cmake/configure.cmake

```diff
@@ -124,6 +125,19 @@ check_type_size("long" SIZEOF_LONG)
 check_type_size("long long" SIZEOF_LONG_LONG)
 check_type_size("size_t" SIZEOF_SIZE_T)
 
+set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
```

Comment:
This will cause the `_GNU_SOURCE` define to be set for _all_ platforms, right now we just set it on Linux/Android in src/mono/CMakeLists.txt.

I think we should do what we do with `CMAKE_REQUIRED_FLAGS` in src/libraries/Native/Unix/configure.cmake for, namely capturing the current content into a temporary variable and then restoring it after the check.

That would also be a better approach than writing to a temporary file and using `try_compile` like we do a few lines below for checking `GLIBC_HAS_CPU_COUNT` (would be nice to replace it as well).

Reply:
I noticed these disparities but didn't tried too hard to match how libs and mono define flags.
In `src/libraries/Native/Unix/configure.cmake` the reason why we do not explicitly set ` _GNU_SOURCE` is that in `src/libraries/Native/Unix/CMakeLists.txt`, we append it to `CMAKE_C_FLAGS` which automagically becomes available to try_compile. In mono, we use `add_definitions` which does not add compiler options to try_compile.